### DISCUSSION TITLE
i3wm + Alacritty preview-tui is working, but cannot disable focus on new window

### DISCUSSION THREAD

**Discussion by ivan94fi** [jarun/nnn#1909] [Q&A]
Hi, I am trying to use preview-tui with i3wm and Alacritty, and the preview window correctly opens and works as expected.

However, I tried to follow the instructions in preview-tui to [disable focus on the preview window](https://github.com/jarun/nnn/blob/86d883eee4986c7125bd77415e1a9b6b94e6bb83/plugins/preview-tui#L94-L95), and it does not work.

I set up NNN with the following options:
```
export NNN_PLUG='p:preview-tui'
```
And configure the terminal as following:
```
export NNN_TERMINAL='alacritty --title preview-tui'
```
Then, in i3 config file:
```
no_focus [title="prevew-tui"]
```

With this configuration, a preview window opens but it is focused as soon as it opens. I also tried to make the preview window floating
```
for_window [title="preview-tui"] floating enable
```
but it is still focused.

If I use the class to identify the preview window, and make it floating, then the floating preview is not focused:
```
# in .bashrc
export NNN_TERMINAL='alacritty --class alacritty_preview'
# in .config/i3/config
for_window [class="alacritty_preview"] floating enable
no_focus [class="alacritty_preview"]
```
However I would prefer to have the preview in a tiled window, but I did not manage to have it in a tiled window and not focused it on open.

Has anyone managed to do this? Thank you.

**Comment by HuruGuru** [comment]
Dunno how, but it works for me
```
# nnn preview
no_focus [class="Alacritty" title="preview-tui"]
focus_on_window_activation none
```

**Reply by ivan94fi** [reply]
Thanks, this actually works. It seems that `focus_on_window_activation none` does the trick. This is actually unexpected for me, as per i3wm's docs it should only control preexisting windows that are somehow activated and request focus. The option to control new windows should be `no_focus`.

I can also confirm that it is not a problem of criteria, because the same criterion is able to modify borders for the preview window, for example.

Maybe it should be reported as a bug in i3wm.

**Reply by ivan94fi** [reply]
Or maybe everything is working as intended and what happens is that the window is first created, it is not focused as per the no_focus rule, but then the preview window is fed data and at that point it requests focus and, if focus_on_window_activation is not disabled, the preview window ultimately gets focus.

I think this is what is actually going on, but it is too fast to see.
